### PR TITLE
Add MTBKeys MTB60 Keyboard (Solderable and Hotswap edition)

### PR DESCRIPTION
**Pull request by kgieselman**
## Description

This PR includes the initial keyboard files for the mtb60, both
editions. In the directory under mtbkeys/mtb60 there are two folders,
one for each edition.

Edtions:
1. Solder - Firmware for the flexible solder version of the MTB60 PCB.
2. Hotswap - Firmware for the hotswap version of the MTB60 PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mtbkeys/mtb60/hotswap/config.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#define RGB_DI_PIN D4
+
+#ifdef RGB_DI_PIN
+#   define RGBLED_NUM 16
+#   define RGBLIGHT_HUE_STEP 8
+#   define RGBLIGHT_SAT_STEP 8
+#   define RGBLIGHT_VAL_STEP 8
+#   define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+#   define RGBLIGHT_SLEEP         /* Turn RGB light off when the host goes to sleep */
+#   define RGBLIGHT_ANIMATIONS     /* Enable all RGB animations */
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/hotswap/config.h:51]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/mtbkeys/mtb60/hotswap/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE=0,    ///< Default Layer
+    _FN  =1     ///< Function Layer
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/hotswap/keymaps/default/keymap.c:21]:
```suggestion
    _BASE,    ///< Default Layer
    _FN       ///< Function Layer
```

File: keyboards/mtbkeys/mtb60/hotswap/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Details
+
+* Keyboard Maintainer: [MTBKeys](https://mtbkeys.com/)
+* Hardware Supported: *MTB60 PCB (Hotswap edition)*
+* Hardware Availability: [HB60 on MTBKeys Website](https://mtbkeys.com/)
+
+## Building Firmware
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/hotswap/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/mtbkeys/mtb60/hotswap/rules.mk [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/hotswap/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic
```

File: keyboards/mtbkeys/mtb60/solder/config.h [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#define RGB_DI_PIN B0
+
+#ifdef RGB_DI_PIN
+#   define RGBLED_NUM 16
+#   define RGBLIGHT_HUE_STEP 8
+#   define RGBLIGHT_SAT_STEP 8
+#   define RGBLIGHT_VAL_STEP 8
+#   define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+#   define RGBLIGHT_SLEEP         /* Turn RGB light off when the host goes to sleep */
+#   define RGBLIGHT_ANIMATIONS     /* Enable all RGB animations */
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/solder/config.h:51]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/mtbkeys/mtb60/solder/readme.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+## Details
+
+* Keyboard Maintainer: [MTBKeys](https://mtbkeys.com/)
+* Hardware Supported: *MTB60 PCB (Solder edition)*
+* Hardware Availability: [HB60 on MTBKeys Website](https://mtbkeys.com/)
+
+## Building Firmware
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mtbkeys/mtb60/solder/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mtbkeys/mtb60/hotswap/config.h (hunk 1 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/hotswap/config.h]: Suggestion is now implemented

File: keyboards/mtbkeys/mtb60/hotswap/keymaps/default/keymap.c (hunk 2 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/hotswap/keymaps/default/keymap.c]: Suggestion is now implemented

File: keyboards/mtbkeys/mtb60/hotswap/readme.md (hunk 3 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/hotswap/readme.md:11]: Suggestion is now implemented

File: keyboards/mtbkeys/mtb60/hotswap/rules.mk (hunk 4 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/hotswap/rules.mk]: Suggestion is now implemented

File: keyboards/mtbkeys/mtb60/solder/config.h (hunk 5 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/solder/config.h]: Suggestion is now implemented

File: keyboards/mtbkeys/mtb60/solder/readme.md (hunk 6 above)
Comment (kgieselman) [reply keyboards/mtbkeys/mtb60/solder/readme.md:11]: Suggestion is now implemented

**Review by drashna — APPROVED** [review]

File: keyboards/mtbkeys/mtb60/solder/info.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
+{
+    "keyboard_name": "MTB60",
+    "url": "mtbkeys.com",
+    "maintainer": "MTBKeys",
+    "width": 15,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/solder/info.json:6]:
```suggestion
```

File: keyboards/mtbkeys/mtb60/hotswap/info.json [hunk 8]
```diff
@@ -0,0 +1,74 @@
+{
+    "keyboard_name": "MTB60",
+    "url": "mtbkeys.com",
+    "maintainer": "MTBKeys",
+    "width": 15,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/hotswap/info.json:6]:
```suggestion
```

File: keyboards/mtbkeys/mtb60/hotswap/readme.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
+# MTB60 (Hotswap Edition)
+
+## Details
+
+* Keyboard Maintainer: [MTBKeys](https://mtbkeys.com/)
+* Hardware Supported: *MTB60 PCB (Hotswap edition)*
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/hotswap/readme.md:6]:
```suggestion
* Keyboard Maintainer: [MTBKeys](https://mtbkeys.com/)
* Hardware Supported: MTB60 PCB (Hotswap edition)
```

File: keyboards/mtbkeys/mtb60/hotswap/readme.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+```sh
+    make mtbkeys/mtb60/hotswap:default:flash
+```
+
+QMK example for this keyboard (after setting up your build enivrionment):
+
+```sh
+    qmk compile -kb mtbkeys/mtb60/hotswap -km default
+```
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/hotswap/readme.md:27]:
```suggestion
    make mtbkeys/mtb60/hotswap:default

Flashing example for this keyboard:

    make mtbkeys/mtb60/hotswap:default:flash
```

File: keyboards/mtbkeys/mtb60/solder/readme.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# MTB60 (Solder Edition)
+
+## Details
+
+* Keyboard Maintainer: [MTBKeys](https://mtbkeys.com/)
+* Hardware Supported: *MTB60 PCB (Solder edition)*
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/solder/readme.md:6]:
```suggestion
* Hardware Supported: MTB60 PCB (Solder edition)
```

File: keyboards/mtbkeys/mtb60/solder/readme.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+```sh
+    make mtbkeys/mtb60/solder:default:flash
+```
+
+QMK example for this keyboard (after setting up your build enivrionment):
+
+```sh
+    qmk compile -kb mtbkeys/mtb60/solder -km default
+```
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/solder/readme.md:27]:
```suggestion
    make mtbkeys/mtb60/solder:default

Flashing example for this keyboard:

    make mtbkeys/mtb60/solder:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mtbkeys/mtb60/solder/rules.mk [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/solder/rules.mk:21]:
```suggestion
```

File: keyboards/mtbkeys/mtb60/hotswap/rules.mk [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/mtbkeys/mtb60/hotswap/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by kgieselman** [comment]
@fauxpark Is there anything that I need to do to get this merged in? This is my first QMK and Github PR so I could be missing something.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by kgieselman** [comment]
I now have 2 approvals, what are next steps to complete the PR and merge it into master? Is there anything required from me?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**